tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+            this.rand = new Random();
+
+            // This variable is used to maintain the scheduling of the connection keep-alive.
+            this.refreshRate = SustainedConnectionWorker.this.spec.refreshRateMs();
+
+            // This variable is used to maintain the connection properties.
+            this.props = new Properties();
+            this.props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, SustainedConnectionWorker.this.spec.bootstrapServers());
+            this.props.put(ConsumerConfig.GROUP_ID_CONFIG, this.consumerGroup);
+            this.props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+            this.props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+            this.props.put(ConsumerConfig.FETCH_MAX_BYTES_CONFIG, 1024);
+            WorkerUtils.addConfigsToProperties(
+                    this.props, SustainedConnectionWorker.this.spec.commonClientConf(), SustainedConnectionWorker.this.spec.consumerConf());
+        }
+
+        @Override
+        public void refresh() {
+            try {
+
+                if (this.consumer == null) {
+
+                    // Housekeeping to track the number of opened connections.
+                    SustainedConnectionWorker.this.totalConsumerConnections.incrementAndGet();
+
+                    // Create the consumer and fetch the partitions for the specified topic.
+                    this.consumer = new KafkaConsumer<>(this.props, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+                    List<TopicPartition> partitions = this.consumer.partitionsFor(this.topicName).stream()
+                            .map(partitionInfo -> new TopicPartition(partitionInfo.topic(), partitionInfo.partition()))
+                            .collect(Collectors.toList());
+
+                    // Select a random partition and assign it.
+                    this.activePartition = partitions.get(this.rand.nextInt(partitions.size()));
+                    this.consumer.assign(Collections.singletonList(this.activePartition));
+                }
+
+                // The behavior when passing in an empty list is to seek to the end of all subscribed partitions.
+                this.consumer.seekToEnd(Collections.emptyList());
+
+                // Poll to keep the connection alive, ignoring any records returned.
+                this.consumer.poll(Duration.ofMillis(50));
+
+            } catch (Throwable e) {
+
+                // Set the consumer to be recreated on the next cycle.
+                this.closeQuietly();
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                SustainedConnectionWorker.this.totalConsumerConnections.decrementAndGet();
+                SustainedConnectionWorker.this.totalConsumerFailedConnections.incrementAndGet();
+                SustainedConnectionWorker.log.error("Error while refreshing sustained KafkaConsumer connection", e.getMessage());
```

Comment:
This won't log the error. We want to use the public void error(String msg, Throwable t); overload, so we need to change e.getMessage() -> e

Reply:
Done.